bot/exts/evergreen/cheatsheet.py

```diff
@@ -0,0 +1,93 @@
+import random
+import re
+import typing as t
+from urllib.parse import quote_plus
+
+from discord import Embed
+from discord.ext import commands
+from discord.ext.commands import BucketType, Context
+
+from bot.constants import Channels, Colours, ERROR_REPLIES
+
+ERROR_MESSAGE = """
+Unknown cheat sheet. Please try to reformulate your query.
+
+**Examples**:
+```md
+.cht read json
+.cht hello world
+.cht lambda
+```
+If the problem persists send a message in <#{channel}>
+"""
+
+
+class ChtSh(commands.Cog):
+    """Commands that sends a result of a cht.sh search in code blocks."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @staticmethod
+    def fmt_error_embed() -> Embed:
+        """If the cht.sh search returned 404, overwrite it to send a custom error embed."""
+        embed = Embed(colour=Colours.soft_red)
+        embed.title = random.choice(ERROR_REPLIES)
+        embed.description = ERROR_MESSAGE.format(channel=Channels.dev_contrib)
+        return embed
+
+    def result_fmt(self, url: str, body_text: str) -> t.Tuple[bool, t.Union[str, Embed]]:
+        """Format Result."""
+        if body_text.startswith("#  404 NOT FOUND"):
+            embed = self.fmt_error_embed()
+            return True, embed
+
+        body_space = min(1986 - len(url), 1000)
+
+        if len(body_text) > body_space:
+            description = f"**Result Of cht.sh**\n" \
+                          f"```python\n{body_text[:body_space]}\n" \
+                          f"... (truncated - too many lines)```\n" \
+                          f"Full results: {url} "
+        else:
+            description = f"**Result Of cht.sh**\n" \
+                          f"```python\n{body_text}```\n" \
+                          f"{url}"
+        return False, description
+
+    @commands.command(
+        name="cheat",
+        aliases=("cht.sh", "cheatsheet", "cheat-sheet", "cht"),
+    )
+    @commands.cooldown(1, 10, BucketType.user)
+    async def cheat_sheet(
+            self, ctx: Context, *search_terms: str
+    ) -> None:
+        """
+        Search cheat.sh.
+
+        Gets a post from https://cheat.sh/python/ by default.
+        Usage:
+        --> .cht read json
+        """
+        url = f'https://cheat.sh/python/{quote_plus(" ".join(search_terms))}'
+        headers = {
+            'User-Agent': 'curl/7.68.0'
```

Comment:
Could do: From https://github.com/chubin/cheat.sh/blob/62a4a586877a8a651b1c52f8398ebbe6dafb6f76/bin/app.py#L76 it looks like the `python-requests` user agent is treated the same as the `curl` user agent. I haven't tested this, but it looks like we don't need to set our own headers here?

Reply:
We do otherwise  the content we receive is contains all this escape codes and stuff, I had tried it (sometime ago) so just directly used this right now.

Reply:
Huh, that's weird. Looking at their source it should behave the exact same way.

If this works, and without it, it doesn't then leave it here :)

Reply:
fwiw, this is because the `self.bot.http_session` is a `aiohttp.ClientSession()` so the user agent is `Python/3.8 aiohttp/3.7.3`

Reply:
My PR into cheat.sh was merged (https://github.com/chubin/cheat.sh/pull/280), so you shouldn't have to overwrite the headers now :). See https://discord.com/channels/267624335836053506/635950537262759947/805490234287849513